### ISSUE TITLE
bug: [0.5.12] `max_tokens` automatically revert back to 8192 when users create a new thread

### ISSUE THREAD

**Issue by khanakia** [janhq/jan#4372]
### Jan version

v0.5.12

### Describe the Bug

Jan just realeased a new version but the bug is still there

Ref: https://github.com/janhq/jan/issues/4238

![Screenshot 2024-12-31 at 2 06 02 PM](https://github.com/user-attachments/assets/4544c24e-7e57-4f88-8dec-4367c72db38c)

### What is your OS?

- [X] MacOS
- [ ] Windows
- [ ] Linux

**Comment by imtuyethan** [comment]
@khanakia Can you create a new thread & try again? It works great on my end with Jan `0.5.12`:
https://github.com/user-attachments/assets/c0ac6592-8466-4fd9-bd06-c9f396b90306

Are you sure you have updated to 0.5.12?

**Comment by khanakia** [comment]
I see i just checked again when you created a new thread it by default it shows `8192` and when you actually types something then it automatically sets the max token down to `4092`

It's kind of misleading from User Experience

**Comment by khanakia** [comment]
I just tried it worked once but again stopped working check video 

## Worked the first time

https://github.com/user-attachments/assets/d3d2fc94-dd9c-4749-b07b-c137ac0c9d94

## Stopped Working again

https://github.com/user-attachments/assets/4ee85c79-12db-4da5-980d-f7e941893345

**Comment by khanakia** [comment]
Actually it only change the max token when you click the inside the Token TextInput field otherwise it does not works

**Comment by louis-jan** [comment]
Hi @khanakia Hi @khanakia, could you please check the models folder to see if there are still legacy OpenAI model.json files? If they are there, please remove them. The fix includes bundling remote model manifests inside the app.

**Comment by khanakia** [comment]
Hi renamed the current models folder and it created a new one which is totally empty now.

PS: I have restarted the jan.app In case you will ask so issue is still there.

![Screenshot 2025-01-02 at 1 35 40 PM](https://github.com/user-attachments/assets/b1dac88f-badb-4b86-a32b-8a1751a6949e)

**Comment by khanakia** [comment]
Now i just deleted the whole `~/jan` folder at and restarted the app.

and did setup again the OpenAI  and now it seems to be working

But as you can see on the Right side it still says `max-tokens: 8192`

![Screenshot 2025-01-02 at 1 39 50 PM](https://github.com/user-attachments/assets/8f0fd638-e390-4a11-8b9d-f54b27efda2f)

**Comment by louis-jan** [comment]
@khanakia, I see. Can you help me select another model and check if the issue persists? The settings should have persisted to the thread.

**Comment by khanakia** [comment]
Selecting the model from Select Box works but when creating a new thread it does updates the max token

Check video please

https://github.com/user-attachments/assets/558e4961-61d0-458b-8f7e-e7d01ebd31b2

**Comment by louis-jan** [comment]
Hey @khanakia, in your video, when creating a new thread, it selects o1-preview, which supports up to 32k.

**Comment by khanakia** [comment]
Yes I know i was pointing out in the video that every time you create a new thread max tokens switch to `8192` 

Check this one

https://github.com/user-attachments/assets/dbd2ac9b-f3ac-4229-abb7-72d61342e2d0

**Comment by imtuyethan** [comment]
This is true @khanakia i can reproduce, thanks for pointing it out!!

**Comment by imtuyethan** [comment]
@louis-jan I can reproduce now:

https://github.com/user-attachments/assets/130beadd-aac1-4f85-b4f9-b9153e22d363

**Comment by louis-jan** [comment]
Thanks @khanakia for pointing out that issue, I can reproduce it here. We'll likely have the fix in the hotfix next week.

**Comment by khanakia** [comment]
Thanks @louis-jan

**Comment by imtuyethan** [comment]
Fixed in the latest beta build, will release next monday:

https://github.com/user-attachments/assets/3ce3c8e6-966a-4395-a336-76d3f5572a72
